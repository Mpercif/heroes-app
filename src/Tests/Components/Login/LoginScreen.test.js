import { mount } from "enzyme"
import { AuthContext } from "../../../auth/AuthContext"
import { LoginScreen } from "../../../Components/Login/LoginScreen"



describe('Pruebas en el componente <LoginScreen />', () => {

    const contextDefault = {
        dispatch: jest.fn()
    }

    const historyMock = {
        replace: jest.fn()
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextDefault}>
            <LoginScreen
                history={historyMock}
            />
        </AuthContext.Provider>
    )
       
    test('Debe mostrarse correctamente ', () => {
        
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de realizar el dispatch y la navegaciÃ³n ', () => {
        
        const handleClick = wrapper.find('button').prop('onClick')
        handleClick()
        expect(contextDefault.dispatch).toHaveBeenCalled()
        expect(historyMock.replace).toHaveBeenCalledWith('/')

        localStorage.setItem('lastPath', '/dc')
        handleClick()
        expect(historyMock.replace).toHaveBeenCalledWith('/dc')
    }) 
})
