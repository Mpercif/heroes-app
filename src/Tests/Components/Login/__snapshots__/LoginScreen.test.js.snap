// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <LoginScreen /> Debe mostrarse correctamente  1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-outline-success"
    onClick={[Function]}
  >
    Ingresar
  </button>
</div>
`;
